src/eos-util.c

```diff
@@ -282,9 +293,29 @@ eos_updater_get_booted_deployment_from_loaded_sysroot (OstreeSysroot *sysroot,
   booted_deployment = ostree_sysroot_get_booted_deployment (sysroot);
   if (booted_deployment == NULL)
     {
-      g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
-                           "Not an ostree system");
-      return NULL;
+      g_autoptr(GPtrArray) deployments = NULL;
+      static OstreeDeployment *fake_booted_deployment = NULL;
+
+      if (!fallback_to_the_first_deployment ())
+        {
+          g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
+                               "Not an ostree system");
+          return NULL;
+        }
+
+      if (fake_booted_deployment == NULL)
+        {
+          deployments = ostree_sysroot_get_deployments (sysroot);
+          if (deployments->len == 0)
+            {
+              g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
+                                   "No deployments found at all");
+              return NULL;
+            }
+          fake_booted_deployment = g_object_ref (g_ptr_array_index (deployments, 0));
+        }
+
+      return g_object_ref (fake_booted_deployment);
```

Comment:
I added that only for testing purposes. During the tests I can setup an ostree sysroot without any problem. The problem is getting the booted deployment, because libostree does some checks if sysroot is actually mounted to `/`. Also, to mock a deployment I would need to introduce more envvars to produce it. So I don't think it is worthwhile.

But will split it to a separate function.